modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/EmptyBucketUserAssignmentForBatchAssignment.java

```diff
@@ -0,0 +1,212 @@
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.intuit.wasabi.tests.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.intuit.wasabi.tests.library.TestBase;
+import com.intuit.wasabi.tests.library.util.Constants;
+import com.intuit.wasabi.tests.model.Application;
+import com.intuit.wasabi.tests.model.Assignment;
+import com.intuit.wasabi.tests.model.Bucket;
+import com.intuit.wasabi.tests.model.Experiment;
+import com.intuit.wasabi.tests.model.User;
+import com.intuit.wasabi.tests.model.factory.ApplicationFactory;
+import com.intuit.wasabi.tests.model.factory.BucketFactory;
+import com.intuit.wasabi.tests.model.factory.ExperimentFactory;
+
+/**
+ * Created on 1st march 2017
+ * This test class tests scenario where a user is assigned to a batch of experiments
+ * and the assigned buckets are deleted to verify that when the assignment call is made
+ * to the same user, then the user should be assigned to one of the available buckets 
+ * in each of the experiments... to keep things simple creating two experiments with
+ * two buckets for each experiment 
+ *
+ */
+public class EmptyBucketUserAssignmentForBatchAssignment extends TestBase {
+
+    Application batchAssignmentEmptyBucketTestApp = null; // app under which the test experiments will be created
+
+    List<Experiment> testExperimentsList = null; // holds experiments that are created in the test class to be deleted eventually in the tearDown
+
+    List<String> assignedBucketLabels = null; // holds the labels of the buckets to which the test user has been assigned
+
+    @BeforeClass
+    public void setUp()
+    {
+        //lets create application and assign it to the variable
+        batchAssignmentEmptyBucketTestApp = ApplicationFactory.createApplication().setName("batchAssignmentEmptyBucketTestApp");
+
+
+        // initialize the declared reference variable 
+       
+        testExperimentsList = new ArrayList<Experiment>();
+
+    }
+
+    /**
+     * This test case tests scenario where we do batch assignment of a user to
+     * list of experiments, delete the assigned buckets and do assignment again
+     * for the same user against the same experiments should make a new assignment
+     */
+    @Test
+    public void createBatchAssignmentAndDeleteAssignedBucketTest() {
```

Comment:
Should this method name be createBatchAssignmentAndEmptyAssignedBucketTest() ? different is EMPTY rather than DELETE as for bucket only following states are valid OPEN, EMPTY, CLOSED.